### ISSUE TITLE
Motion sensor stuck on "detected" on Arlo video bell

### ISSUE THREAD

**Issue by Kilmanagh** [twrecked/hass-aarlo#645]
I have two Arlo Wired Video DoorBell Cameras that I am using on Home assistant. Both the same models (AVD1001), although The SKU's are slightly different. 
**Front Door: AVD1001W-100NAS**   **Office Door: AVD1001-100NAS**
 

- I have a paid Arlo subscription for the devices.
- both cameras properly send notification detections in the Arlo App on my phone
- notifications are configured identical in device options on the Arlo App.
- lovelace cards are identical to their assigned entities. 
- Both cameras appear on Arlo Lovelace correctly with one exception:
- **Front Door** is always **Detected**, (it seems to be in that state from HA boot and remains there). The Arlo app is reporting properly as clear. Otherwise 
- **Office Door** states are working perfectly; I can use them for automations in HA. 

If anyone needs further information, please let me know. I am stumped at the moment. I have seen other bugs on this issue, but i wanted to present some more detail than what I have read so far.

**Comment by twrecked** [comment]
Can you turn on and paste some debug.
 [see here](https://github.com/twrecked/hass-aarlo#enabling-and-examining-logs) for the logging
 [see here](https://github.com/twrecked/hass-aarlo#hiding-sensitive-data) on how to encrypt it so only I can see it

**Comment by Kilmanagh** [comment]
I am sifting through the log before I post it in a bit. It is over 3m long and I need to vet the data.

A few things I noticed: **Front Door**: battery level is always **38%** , Office Door 49%. I am not sure where they are getting those as they are both wired to a separate transformer and do not contain batteries at all. 
Secondly, it will not PING to the front door, it doesn't even call it by name, but by its serial number but repeats that it is a No Ping. Office Door is fine.

for example:

[pyaarlo] pinging Office Door
[pyaarlo] NO ping to A2G207ZRA0CF9

I am not what they are referring to "ping" but the IP addresses PING just fine on my network.

**Comment by twrecked** [comment]
The battery level will appear in the log as `batteryLevel`, if that matches with the 38% or the 49% then my code is just parroting the number Arlo sends me. If these are wired I may have to alter the code to not create the devices.

And it's not a real `ping`, we just make a request to the Arlo servers for a device status update. We don't send it for all devices and that log just helps me identify which we ignore. I actually have no idea of the IP address of the Arlo devices, the code talks directly with their servers.

And I appreciate the logs are long, one thing to look for would be to trigger the motion and look for `motionDetected`.

**Comment by Kilmanagh** [comment]
I triggered it and captured it. The device is XXXXXXXXXCB9, there is no motionDetected (even no Falses) buy plenty of motionRecord. The Arlo app updated at once, but nothing in HA. My logs from earlier show the other Arlo reporting motionDetected.

**Comment by twrecked** [comment]
I'll take a look at the logs. Does the HA Arlo account have `admin access`?

**Comment by twrecked** [comment]
I think it might be related to the ping not being sent to that particular device, I think I might need to update my `is_corded` check that looks for power supplies. If you kept the logs can you look near the start up some packets from Arlo that look like this:

```
{ 'action': 'is',
  'from': 'XXXXXXXXXX',
  'properties': [ { 'activityState': 'idle',
                    'audioDetected': False,
                    'audioMicAGC': 0,
                    'batteryLevel': 81,
                    'batteryTech': 'Rechargeable',
                    'bestLocalLiveStreaming': 'off',
                    'blockNotifications': { 'block': False,
                                            'duration': 0,
                                            'endTime': 0},
                    'brightness': 0,
                    'chargeNotificationLedEnable': True,
                    'chargeOnly': False,
                    'chargerTech': 'None',
                    'chargingState': 'Off',
                    'connectionState': 'available',
                    'continuousStreamState': 'inactive',
                    'flip': False,
                    'hasStreamed': True,

```

You have to look at the `serialNumber` field to find the device we are interested in. I need the `battery*` and `charger*` fields, or anything that might indicate the device is hard wired.

**Comment by twrecked** [comment]
That's the URL I'm using to get the information.

Can you try logging into the [Arlo camera](https://my.arlo.com/#/login) web site with the Home Assistant account and check the **Front Door** in there? You will need to stop HA while you do this.

**Comment by Kilmanagh** [comment]
Yes it looks fine when I log in with a web browser. I did a restart with the device hoping it would make a difference, but it did not. 
The funny thing is that Last Motion is always correct, but I think it is getting that from API that points to the dashboard, not the device data. I am wondering if it is the API service or perhaps it isnt liking the call to the device. I still do not see any response to the Api call. I just have no idea at this point.

**Comment by twrecked** [comment]
I think Arlo changed the back end. Let me play around here and I'll push an update to try.

**Comment by twrecked** [comment]
Can you try the latest release. #646 may have fixed the issue.

**Comment by Kilmanagh** [comment]
I tried it to no avail. What I did do is install another Arlo that was new in the box to see it was a device issue. The new device has the same problems. What I did figure out, this new device reports 36% battery life in HA. I swapped it back out and lo and behold despite it being unplugged it is still powered and it was still sending notifications and snapshots to Arlo for about 20 minutes. So these are wired doorbells with a small battery if the power goes out for a little bit. 

Does the code suppress the pings and other functions if it thinks the battery is low? The battery level is never going to show very high. I am not sure why the one has been working, but it always shows 48% instead of 38 or 36%

**Comment by twrecked** [comment]
We suppress the pings to some devices because, in the past, we had problems with rapid battery usage. In your case you're pinging the _Office Door_ so why not the _Front Door_. It has to be related to the `Front Door` not coming back from the JSON call. Can you send the logs from when you do see the _Font Door_ and _Office Door_ first appear in a response from Arlo? I want see if there is an obvious state different between the two. 

Any chance your comfortable with web browser debug?

In the meantime,  if you are ok with an editor? Or using _patch_. One thing to try would be to turn on the pings. You can either apply the following diff or change `False` to `True` on line 486 on `pyaarlo/base.py`

```diff
diff --git a/custom_components/aarlo/pyaarlo/base.py b/custom_components/aarlo/pyaarlo/base.py
index 037be52..85b6d97 100644
--- a/custom_components/aarlo/pyaarlo/base.py
+++ b/custom_components/aarlo/pyaarlo/base.py
@@ -483,7 +483,7 @@ class ArloBase(ArloDevice):
             if self.model_id.startswith(MODEL_BABY):
                 return True
             if self.is_own_parent and self.using_wifi and not self.is_corded:
-                return False
+                return True
             # Don't ping these devices ever.
             if self.model_id.startswith(
                 (
```

**Comment by Kilmanagh** [comment]
Yes, I am comfortable with debugging just about any code as a long-time developer. Python is something i have not tackled yet but looks quite painless.

So, returning the code at 486 as true, not only allows the ping to work, but fixes the whole problem! They all work correctly now.
I set a few new debug outputs of my own to test a few things. I am trying to figure out how you are finding the value 'self.is_corded' from the initial API call? There is only indication that there is Wifi or not. So we will always get a false if the code initialized as false. I am guessing iit might be the url=https://myapi.arlo.com/hmsweb/v2/users/devices?t=XXXXXXXXXXXXXX query?
After that it seems to never bother to check those properties 'properties': [ { 'activityState': 'idle' -- etc. to see if it is corded or not. it simply ignores the device properties and changes of state. For some reason the **FIRST** device seems to be an exception.
Now that it can read the device, the battery level is now over 50%. 

In short, making L: 486 true = success! Maybe they changed the API response on you for is_corded and put it inside the device dataset instead?

**Comment by twrecked** [comment]
Python is pretty easy.  The problem I have is I'm having to reverse engineer the protocol with only 1/5 of the different types of equipment at my house so it's good to know you don't mind playing about with the code,

`is_corded` should update as new information comes in. This can either come from the device URL (`https://myapi.arlo.com/hmsweb/v2/users/devices?t=XXXXXXXXXXXXXX`) or the event stream directly from Arlo. For example, I have a solar panel attached to one of my cameras and `charging` will change to `True` if (when) the sun comes out. The method is badly named, I think `is_corded_only` would be better.

Can you checkout the current values of _Front Door_ by going to `Developer Tools --> States` and looking for that camera entry. These are the fields I'm interested in:

```
    battery_level: 98
    battery_tech: Primary
    ...
    charging: false
    charger_type: None
    wired: false
    wired_only: false
    ...
    wifi: false
    corded: false
    ...
    model_id: VMC3030
```

With this information I'll see if I can fix the logic. It could be as simple as adding in a charger check like this:

```diff
diff --git a/custom_components/aarlo/pyaarlo/base.py b/custom_components/aarlo/pyaarlo/base.py
index 037be52..5c74138 100644
--- a/custom_components/aarlo/pyaarlo/base.py
+++ b/custom_components/aarlo/pyaarlo/base.py
@@ -482,7 +482,7 @@ class ArloBase(ArloDevice):
             # Battery powered wifi devices that act as their own base station don't get pinged.
             if self.model_id.startswith(MODEL_BABY):
                 return True
-            if self.is_own_parent and self.using_wifi and not self.is_corded:
+            if self.is_own_parent and self.using_wifi and not self.is_corded and not self.has_charger:
                 return False
             # Don't ping these devices ever.
             if self.model_id.startswith(
```

**Comment by Kilmanagh** [comment]
What I see is inconsistency in the two cameras that are the same model and setup. Both are on the WIFI, both are powered by their own AC transformers. 
 
model_name: AVD1001A
brand: Netgear Arlo
battery_level: 59
battery_tech: Rechargeable
charging: true
charger_type: VAC Charger
wired: true
wired_only: false
**wifi: false**
corded: false
model_id: AVD1001A
friendly_name: Office Door

model_name: AVD1001A
brand: Netgear Arlo
battery_level: 51
battery_tech: Rechargeable
charging: true
charger_type: VAC Charger
wired: true
wired_only: false
**wifi: true**
corded: false
model_id: AVD1001A
friendly_name: Front Door

**Comment by Kilmanagh** [comment]
Office Door: 'connectivity': {'mepStatus': 'unknown'},         <---------------- nothing supplied 
Front Door: 'connectivity': {'connected': True, 'mepStatus': 'unknown', 'type': 'wifi'},

{ 'action': 'is',
  'from': 'XXXXXXXXXXXXXX',
  'properties': {'connectionState': 'available'},
  'resource': 'basestation',
  'transId': 'XXXXXXXXXXXXXXXXXXXXXXX'}

and some unhandled response basestation - {'from': 'XXXXXXXX', 'action': 'is', 'resource': 'basestation', 'properties': {'connectionState': 'available'}, 'transId': 'XXXXXXXXXXXXXXXXXXXX'} but that may be due to the fact I disconnected that extra camera.

**Note: basestation NEVER supplies anything but connectionstate. signalStrength, etc is supplied by the camera resource**
